### PR TITLE
Add type hints to `test_requests.py`

### PR DESCRIPTION
**Pull request by TechNiick**
# Summary

Related to [this](https://github.com/encode/starlette/pull/2445)
Type annotation added to `test_requests.py`

# Checklist

- [ ] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

File: tests/test_requests.py [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
 
-def test_request_url(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_url(test_client_factory: typing.Callable[..., TestClient]) -> None:
+    async def app(
+        scope: typing.Dict[str, typing.Any],
+        receive: typing.Callable[[], typing.Awaitable[typing.Dict[str, typing.Any]]],
+        send: typing.Callable[
+            [typing.MutableMapping[str, typing.Any]], typing.Awaitable[None]
+        ],
```
Comment (Kludex) [inline tests/test_requests.py]: Why are you not using `Scope`, `Receive`, and `Send`?

File: tests/test_requests.py [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
 from starlette.requests import ClientDisconnect, Request
 from starlette.responses import JSONResponse, PlainTextResponse, Response
-from starlette.types import Message, Scope
+from starlette.testclient import TestClient
+from starlette.types import (
+    Message,
+    Receive,
+    Scope,
+    Send,
+)
```
Comment (Kludex) [inline tests/test_requests.py]: Please fix those.
```suggestion
from starlette.types import Message, Receive, Scope, Send
```

**Review by Kludex — CHANGES_REQUESTED** [review]

File: tests/test_requests.py (hunk 1 above)
Comment (TechNiick) [reply tests/test_requests.py]: Updated those in all places in this file.

Just a small note:
There are some places where I have added: `test_client_factory: typing.Any`. In those places I get an error when i try to use the `TestClientFactory`:
example: `Argument "data" to "post" of "TestClient" has incompatible type "str"; expected "Optional[Mapping[str, Union[str, Iterable[str]]]]"`

File: tests/test_requests.py [hunk 3]
```diff
@@ -128,8 +132,12 @@ async def app(scope, receive, send):
...
 
 
-def test_request_form_context_manager(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_form_context_manager(test_client_factory: TestClientFactory) -> None:
+    async def app(
+        scope: Scope,
+        receive: Receive,
+        send: Send,
+    ) -> None:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
    async def app(scope: Scope, receive: Receive, send: Send) -> None:
```

File: tests/test_requests.py [hunk 4]
```diff
@@ -232,17 +242,20 @@ async def app(scope, receive, send):
...
+    anyio_backend_name: str,
+    anyio_backend_options: typing.Dict[str, typing.Any],
+) -> None:
     """
     If a client disconnect occurs while reading request body
     then ClientDisconnect should be raised.
     """
 
-    async def app(scope, receive, send):
+    async def app(scope: Scope, receive: Receive, send: typing.Any) -> None:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
    async def app(scope: Scope, receive: Receive, send: Send) -> None:
```

File: tests/test_requests.py [hunk 5]
```diff
@@ -232,17 +242,20 @@ async def app(scope, receive, send):
...
         request = Request(scope, receive)
         await request.body()
 
-    async def receiver():
+    async def receiver() -> typing.Dict[str, typing.Any]:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
    async def receiver() -> Message:
```

File: tests/test_requests.py [hunk 6]
```diff
@@ -141,8 +149,8 @@ async def app(scope, receive, send):
     assert response.json() == {"form": {"abc": "123 @"}}
 
 
-def test_request_body_then_stream(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_body_then_stream(test_client_factory: typing.Any) -> None:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
def test_request_body_then_stream(test_client_factory: TestClientFactory) -> None:
```

File: tests/test_requests.py [hunk 7]
```diff
@@ -157,8 +165,8 @@ async def app(scope, receive, send):
     assert response.json() == {"body": "abc", "stream": "abc"}
 
 
-def test_request_stream_then_body(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_stream_then_body(test_client_factory: typing.Any) -> None:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
def test_request_stream_then_body(test_client_factory: TestClientFactory) -> None:
```

**Review by Kludex — CHANGES_REQUESTED** [review]

File: tests/test_requests.py [hunk 8]
```diff
@@ -1,17 +1,21 @@
 import sys
-from typing import List, Optional
+import typing
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
```

File: tests/test_requests.py [hunk 9]
```diff
@@ -69,14 +73,14 @@ async def app(scope, receive, send):
...
-def test_request_client(scope: Scope, expected_client: Optional[Address]):
+def test_request_client(scope: Scope, expected_client: Optional[Address]) -> None:
     scope.update({"type": "http"})  # required by Request's constructor
     client = Request(scope).client
     assert client == expected_client
 
 
-def test_request_body(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_body(test_client_factory: typing.Any) -> None:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
def test_request_body(test_client_factory: TestClientFactory) -> None:
```

File: tests/test_requests.py [hunk 10]
```diff
@@ -94,8 +98,8 @@ async def app(scope, receive, send):
     assert response.json() == {"body": "abc"}
 
 
-def test_request_stream(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_stream(test_client_factory: typing.Any) -> None:
```
Comment (Kludex) [inline tests/test_requests.py]:
```suggestion
def test_request_stream(test_client_factory: TestClientFactory) -> None:
```

**Review by Kludex — CHANGES_REQUESTED** [review]

**Review by Kludex — APPROVED** [review]

**Pull request merged**